proxy_manager.go

```diff
@@ -64,10 +64,8 @@ func (m *proxyManager) Proxies() map[string]interface{} {
 	return cp
 }
 
-func (m *proxyManager) getMap(ctx context.Context, name string) (*Map, error) {
-	p, err := m.proxyFor(ctx, ServiceNameMap, name, func(p *proxy) (interface{}, error) {
-		return newMap(p), nil
-	})
+func (m *proxyManager) getMap(ctx context.Context, name string, fn func(p *proxy) (interface{}, error)) (*Map, error) {
+	p, err := m.proxyFor(ctx, ServiceNameMap, name, fn)
```

Comment:
What we are doing is effectively the same thing but different from other getter methods.(getQueue, getTopic etc.) Are you aware of that?